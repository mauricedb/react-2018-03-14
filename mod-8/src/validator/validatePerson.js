export default person => {
  const errors = [];

  if (!person) {
    errors.push('Geen persoon!');
  } else {
    if (!person.firstName) {
      errors.push('Voornaam is verplicht!');
    }

    if (person.firstName.length < 2) {
      errors.push('Voornaam is te kort!');
    }

    if (!person.lastName) {
      errors.push('Achternaam is verplicht!');
    }

    if (person.lastName.length < 2) {
      errors.push('Achternaam is te kort!');
    }

    if (person.age < 0) {
      errors.push('Moet al geboren zijn!');
    }

    if (person.age > 120) {
      errors.push('Zo oud wordt niemand!');
    }
  }

  return errors;
};
