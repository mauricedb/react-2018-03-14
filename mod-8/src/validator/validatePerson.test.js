import validatePerson from './validatePerson';

describe('People validation', () => {
  let person;

  beforeEach(() => {
    person = {
      firstName: 'Li',
      lastName: '12',
      age: 50
    };
  });

  test('Validate an null object', () => {
    const errors = validatePerson(null);
    expect(errors).toEqual(['Geen persoon!']);
  });

  test('Validate a valid object', () => {
    const errors = validatePerson(person);
    expect(errors).toEqual([]);
  });

  test('Validate a over aged', () => {
    person.age = 200;
    const errors = validatePerson(person);
    expect(errors).toEqual(['Zo oud wordt niemand!']);
  });

  test('Validate an unnamed person', () => {
    person.firstName = '';
    const errors = validatePerson(person);
    expect(errors).toEqual(['Voornaam is verplicht!', 'Voornaam is te kort!']);
  });

  test('An unnamed person should match its snapshot', () => {
    person.firstName = '';
    const errors = validatePerson(person);
    expect(errors).toMatchSnapshot();
  });

  test('An valid person should match its snapshot', () => {
    const errors = validatePerson(person);
    expect(errors).toMatchSnapshot();
  });
});

describe('Adding (math)', () => {
  test('0.1 + 0.2 === 0.3', () => {
    expect(0.1 + 0.2).toBeCloseTo(0.3);
  });
});
