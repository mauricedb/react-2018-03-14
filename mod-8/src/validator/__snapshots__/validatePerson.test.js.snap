// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People validation An unnamed person should match its snapshot 1`] = `
Array [
  "Voornaam is verplicht!",
  "Voornaam is te kort!",
]
`;

exports[`People validation An valid person should match its snapshot 1`] = `Array []`;
